### PR TITLE
Iterator

### PR DESCRIPTION
**Pull request by zeina1i**

### CODE REVIEW

File: iterator.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		if item, err := df.ReadAt(int(record.H.DataPos)); err == nil {
+			err = df.rwManager.Release()
+			if err != nil {
+				return false, err
+			}
+
+			it.entry = item
+			return true, nil
+		} else {
+			err := df.rwManager.Release()
```
Comment (xujiajun) [inline iterator.go:95]: This error will override the error returned by ReadAt above

File: iterator.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+	current *Node
+	i       int
+
+	bucket string
+
+	entry *Entry
+}
+
+func newIterator(tx *Tx, bucket string) *Iterator {
```
Comment (xujiajun) [inline iterator.go:30]: This should be a public method？ newIterator =》NewIterator ？ right？

**Review by xujiajun — COMMENTED** [review]
I gave some suggestions, you can take a look

File: iterator.go (hunk 1 above)
Comment (zeina1i) [reply iterator.go]: We have this problem in some other places as well. I fixed all places that had this problem.

File: iterator.go (hunk 2 above)
Comment (zeina1i) [reply iterator.go]: Oh, I made a mistake. Thanks for noticing. Fixed.

File: iterator.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		if item, err := df.ReadAt(int(record.H.DataPos)); err == nil {
+			err = df.rwManager.Release()
+			if err != nil {
+				return false, err
+			}
+
+			it.entry = item
+			return true, nil
+		} else {
+			err2 := df.rwManager.Release()
```
Comment (xujiajun) [inline iterator.go:95]: Variable name with number (err2) doesn't look very good, maybe releaseErr is better?

Comment (zeina1i) [reply iterator.go]: Hey @xujiajun. That's right. I just Fixed that.

**Review by xujiajun — APPROVED** [review]
👍🏻 LGTM

**Pull request merged**